### PR TITLE
Added context generator for logrotate

### PR DESCRIPTION
**Pull request by dparv**
New simple context generator for logrotate + template.

The logrotate context generator has 3 arguments:
 * interval
 * rotate count
 * location of log files in /var/log/

Added coresponding template.

Related-Bug: #1692118

### CODE REVIEW

**Comment by dparv** [comment]
Pull-request updated, HEAD is now 608096ebd63d87ce9983ce985359f779fde23fab

File: charmhelpers/contrib/openstack/context.py [hunk 1]
```diff
@@ -1935,3 +1935,20 @@ def __call__(self):
         return {
             'openstack_release': ostack,
             'operating_system_release': osystem}
+
+
+class LogrotateContext(OSContextGenerator):
+    """Common context generator for logrotate."""
+
+    def __init__(self, location=None, interval=None, count=None):
```
Comment (ajkavanagh) [inline charmhelpers/contrib/openstack/context.py]: Should these values be `None`.  I suspect that, according to the example, the signature should be something like:

```python
    def __init__(self, location, interval=4, count=4):
```

This function also needs a docstring to describe the parameters and their meaning. e.g.

```python
    :param location: Absolute path for the log rotate config file
    :type location: str
    :param interval: The interval between rotations in x units (default 4)
    :type interval: int
    :param count: The number of log files to keep (default 4)
    :type count: int
```

File: charmhelpers/contrib/openstack/context.py [hunk 2]
```diff
@@ -1935,3 +1935,20 @@ def __call__(self):
...
+        self.location = location
+        self.interval = interval
+        self.count = ('rotate %s' % count)
```
Comment (ajkavanagh) [inline charmhelpers/contrib/openstack/context.py]: This should use `.format()` please.

**Comment by dparv** [comment]
Pull-request updated, HEAD is now 45fd83ec20390cb0a462d0b08e32a303b35f4da4

**Comment by dparv** [comment]
Pull-request updated, HEAD is now a2bf593f473c95922e2b1d277bece3f9415a7da2

**Pull request merged**